File: app/src/main/java/dnd/combattracker/adapters/EncounterAdapter.java
package dnd.combattracker.adapters;

import android.database.Cursor;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import dnd.combattracker.R;
import dnd.combattracker.adapters.viewholders.EncounterViewHolder;
import dnd.combattracker.listeners.OnItemClickListener;
import dnd.combattracker.listeners.OnLongItemClickListener;

public class EncounterAdapter extends CursorAdapter<EncounterViewHolder> implements OnItemClickListener, OnLongItemClickListener {

    private OnItemClickListener listener = null;
    private OnLongItemClickListener longListener = null;

    public EncounterAdapter(Cursor cursor) {
        super(cursor);
    }

    @Override
    public EncounterViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.list_item_encounter, parent, false);

        return new EncounterViewHolder(v);
    }

    public void onBindViewHolder(EncounterViewHolder holder, Cursor cursor) {
        cursor.moveToPosition(cursor.getPosition());

        holder.setData(cursor);
        holder.setOnItemClickListener(this);
        holder.setOnLongItemClickListener(this);
    }

    @Override
    public void onItemClick(View itemView, int position) {
        if (listener != null) {
            listener.onItemClick(itemView, position);
        }
    }

    @Override
    public void onLongItemClick(View itemView, int position) {
        if (longListener != null) {
            longListener.onLongItemClick(itemView, position);
        }
    }

    public void setOnItemClickListener(OnItemClickListener listener) {
        this.listener = listener;
    }

    public void setOnLongItemClickListener(OnLongItemClickListener longListener) {
        this.longListener = longListener;
    }


}

File: app/src/main/java/tiru/initiative/adapters/EncounterAdapter.kt
package tiru.initiative.adapters

import android.database.Cursor

import android.view.ViewGroup
import android.view.LayoutInflater
import tiru.initiative.R

class EncounterAdapter(_cursor: Cursor?) : CursorAdapter<EncounterViewHolder>(_cursor) {
    override fun onBindViewHolder(holder: EncounterViewHolder, cursor: Cursor) {
        cursor.moveToPosition(cursor.position)

        holder.setData(cursor)
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): EncounterViewHolder {
        val v = LayoutInflater.from(parent?.context).inflate(R.layout.list_item_encounter, parent, false)

        return EncounterViewHolder(v)
    }
}

